src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/CollectFunction.cs

```diff
@@ -299,10 +299,9 @@ public virtual async Task<FormulaValue> InvokeAsync(FormulaValue[] args, Cancell
 
             var tableValue = arg0 as TableValue;
             var recordValue = arg1 as RecordValue;
-            var recordValueCopy = FormulaValue.NewRecordFromFields(recordValue.Fields);
```

Comment:
fwiw, this line was added in #1145 
This current PR is effectively reverting 1145. 

Strange that removing it doesn't cause any test failures.

Reply:
What does this revert mean? 
With({r:{Value:5}}, Collect(t1, r); Patch(t1, Last(t1), {Value:-1}); Notify(r.Value))

What will be r.Value now? since its by ref now, I assume -1?
NOTE: PA doesn't copy by ref and will return 5.

Reply:
We are still getting `{Value:5}`. I added these test cases and some extra ones to this PR.

Reply:
When patching, Fx is copying the record here https://github.com/microsoft/Power-Fx/blob/c8ebad07563535b4a12be979a0f6a2f20081b467/src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs#L240.
We are hitting the same problem when executing `Patch`.